src/seccomp.c

```diff
@@ -22,24 +22,317 @@
+	map_add("PR_TASK_PERF_EVENTS_DISABLE",
+		(void *)PR_TASK_PERF_EVENTS_DISABLE);
+	map_add("PR_TASK_PERF_EVENTS_ENABLE",
+		(void *)PR_TASK_PERF_EVENTS_ENABLE);
+	map_add("PR_GET_THP_DISABLE", (void *)PR_GET_THP_DISABLE);
+	map_add("PR_GET_TID_ADDRESS", (void *)PR_GET_TID_ADDRESS);
+	map_add("PR_SET_TIMERSLACK", (void *)PR_SET_TIMERSLACK);
+	map_add("PR_GET_TIMERSLACK", (void *)PR_GET_TIMERSLACK);
+	map_add("PR_SET_TIMING", (void *)PR_SET_TIMING);
+	map_add("PR_GET_TIMING", (void *)PR_GET_TIMING);
+	map_add("PR_SET_TSC", (void *)PR_SET_TSC);
+	map_add("PR_GET_TSC", (void *)PR_GET_TSC);
+	map_add("PR_SET_UNALIGN", (void *)PR_SET_UNALIGN);
+	map_add("PR_GET_UNALIGN", (void *)PR_GET_UNALIGN);
+
+	// man 2 getpriority
+	map_add("PRIO_PROCESS", (void *)PRIO_PROCESS);
+	map_add("PRIO_PGRP", (void *)PRIO_PGRP);
+	map_add("PRIO_USER", (void *)PRIO_USER);
+}
+
+void map_destroy()
+{
+	hdestroy_r(&map_htab);
+}
+
+/* Caller must check if errno != 0 */
+scmp_datum_t read_number(char *s)
+{
+	scmp_datum_t val = 0;
+
+	errno = 0;
+
+	// per seccomp.h definition of scmp_datum_t, negative numbers are not
+	// supported, so fail if we see one or if we get one. Also fail if
+	// string is 0 length.
+	if (s[0] == '-' || s[0] == '\0') {
+		errno = EINVAL;
+		return val;
+	}
+	// check if number
+	for (int i = 0; i < strlen(s); i++) {
+		if (isdigit(s[i]) == 0) {
+			errno = EINVAL;
+			break;
+		}
+	}
+	if (errno == 0) {	// found a number, so parse it
+		char *end;
+		// strtol may set errno to ERANGE
+		val = strtoul(s, &end, 10);
+		if (end == s || *end != '\0')
+			errno = EINVAL;
+	} else			// try our map (map_search sets errno)
+		val = map_search(s);
+
+	return val;
+}
+
+int parse_line(char *line, struct seccomp_args *sargs)
+{
+	// strtok_r needs a pointer to keep track of where it is in the
+	// string.
+	char *buf_saveptr;
+
+	// Initialize our struct
+	sargs->length = 0;
+	sargs->syscall_nr = -1;
+
+	if (strlen(line) == 0)
+		return PARSE_ERROR;
+
+	// Initialize tokenizer and obtain first token.
+	char *buf_token = strtok_r(line, " \t", &buf_saveptr);
+	if (buf_token == NULL)
+		return PARSE_ERROR;
+
+	// syscall not available on this arch/kernel
+	sargs->syscall_nr = seccomp_syscall_resolve_name(buf_token);
+	if (sargs->syscall_nr == __NR_SCMP_ERROR)
+		return PARSE_INVALID_SYSCALL;
+
+	// Parse for syscall arguments. Since we haven't yet searched for the
+	// next token, buf_token is still the syscall itself so start 'pos' as
+	// -1 and only if there is an arg to parse, increment it.
+	int pos = -1;
+	while (pos < SC_ARGS_MAXLENGTH) {
+		buf_token = strtok_r(NULL, " \t", &buf_saveptr);
+		if (buf_token == NULL)
+			break;
+		// we found a token, so increment position and process it
+		pos++;
+		if (strcmp(buf_token, "-") == 0)	// skip arg
+			continue;
+
+		enum scmp_compare op = -1;
+		scmp_datum_t value = 0;
+		if (strlen(buf_token) == 0) {
+			return PARSE_ERROR;
+		} else if (strlen(buf_token) == 1) {
```

Comment:
I'm not checking for '=' here because '=NNN' is not part of the syntax (you just specify 'NNN' instead). This is checking if the token has a length of 1, parse it as a number and let read_number() set errno. The comment tried to mention that, but it used 'syscall 1'-- I'll update that to use 'syscall N' and adjust the comment to make it more clear.